src/vivarium_conic_lsff/components/lbwsg.py

```diff
@@ -47,11 +49,29 @@ def setup(self, builder):
         )
 
     def on_initialize_simulants(self, pop_data):
+        def assign_bw_cat(val):
+            if val < project_globals.UNDERWEIGHT:
+                return project_globals.BIRTH_WEIGHT_UNDERWEIGHT
+            return project_globals.BIRTH_WEIGHT_NORMAL
+
+        def assign_ga_cat(val):
+            if val < project_globals.PRETERM:
+                return project_globals.GESTATIONAL_AGE_PRETERM
+            return project_globals.GESTATIONAL_AGE_NORMAL
+
         exposure = self.exposure_distribution.get_birth_weight_and_gestational_age(pop_data.index)
-        self.population_view.update(pd.DataFrame({
+
```

Comment:
This is incorrect.  The exposure here is the source of the exposure pipeline, not it's output after being modified by interventions.  There's a bunch of confusing stuff going on here.  You'll need to record from `self.exposure(pop_data.index, skip_post_processor=True)` *after* you update the state table.

Reply:
This not totally valid, split in the mortality observer by grabbing the pipeline `low_birth_weight_and_short_gestation.exposure` and calling with kwarg skip_post_processor=True when collecting metrics.  Use the returned birthweight and gestational age to split pop.